implementations/scripts/SourceCode.py

```diff
@@ -0,0 +1,101 @@
+class IsSourceCode:
```

Comment:
If you want to implement this as a class, that's fine with me. But it should be a hierarchy of classes. You can start for example with a naive class (that let's anything be source code), but then you can extend from it (or just provide another class with the same `check()` function that implements for example the heuristics you have here, based on that list of excluded paths. Then, when instantiating the Metric, you would provide as argument one of these classes (or objects made from it).

Reply:
Got it! will do.

Reply:
I could try implementing a simple factory pattern? 
Something like this (names are only for explanatory purposes):
```
class Option():
    check()

class Naive(Option):
    check()

class FileExclude(Option):
    check()
...
..

class ChooseOption():
    # choose right option
    option.check()
```

Reply:
Yes, something like that, except for the `ChooseOption` class, which I don't understand. Usually, you would just pass the intended class as a parameter when instantiating the corresponding object in the Metrics hierarchy, and that object will just call check in that class for each commit...

Reply:
If you look at my latest patch:
```python
class IsSourceCode:

     def __init__(self, source_code_exclude_list=None, algorithm_class=None):
        self.algorithm_obj = globals()[algorithm_class](source_code_exclude_list)

     def check(self, commit):
        """
        Returns the check method of the algorithm to be used. 
        :param commit: a commit structure, returned by the _flatten_data method.
        """
        return self.algorithm_obj.check(commit)
```
it allows the user to enter a string and not have to create an object and pass it to the function. The algorithm classes are hidden from the user, so to speak.

Reply:
Also, please have a look at my comment [here](https://github.com/chaoss/wg-evolution/pull/162#discussion_r293298796): I have a doubt regarding the `flatten` method.